ui/app/components/dashboard/quick-actions-card.js

```diff
@@ -0,0 +1,182 @@
+        buttonText: 'Issue leaf certificate',
+        model: 'pki/role',
+        path: 'vault.cluster.secrets.backend.pki.roles.role.generate',
+      };
+    case 'view-certificate-pki':
+      return {
+        title: 'Certificate serial number',
+        placeholder: '33:a3:...',
+        elementType: 'search-select',
+        buttonText: 'View certificate',
+        model: 'pki/certificate/base',
+        path: 'vault.cluster.secrets.backend.pki.certificates.certificate.details',
+      };
+    case 'view-issuer-pki':
+      return {
+        title: 'Issuer',
+        placeholder: 'Type issuer name or ID',
+        elementType: 'search-select',
+        buttonText: 'View issuer',
+        model: 'pki/issuer',
+        nameKey: 'issuerName',
+        path: 'vault.cluster.secrets.backend.pki.issuers.issuer.details',
+      };
+  }
+};
+
+export default class DashboardQuickActionsCard extends Component {
+  @service router;
+
+  @tracked selectedEngine;
+  @tracked selectedActions = [];
+  @tracked selectedAction;
+  @tracked actionParamField;
+  @tracked selectedEngineName;
+  @tracked value;
+
+  constructor() {
+    super(...arguments);
+
+    if (!this.selectedEngine) {
+      this.setSelectedAction();
+    }
+  }
+
+  get filteredSecretEngines() {
+    return this.args.secretsEngines.filter(
+      (secretEngine) =>
+        (secretEngine.shouldIncludeInList && secretEngine.type === 'pki') ||
+        secretEngine.type === 'kv' ||
+        secretEngine.type === 'database'
+    );
+  }
+
+  get secretsEnginesOptions() {
+    return this.filteredSecretEngines.map((filteredSecretEngine) => {
+      const name = filteredSecretEngine.path.replace('/', '');
+
+      return {
+        name,
+        id: `${filteredSecretEngine.type} ${name}`,
+      };
+    });
+  }
+
+  @action
+  handleSearchEngineSelect([selectedSearchEngines]) {
+    if (selectedSearchEngines?.length) {
+      const selectedEngine = selectedSearchEngines.split(' ');
+      this.selectedEngine = selectedEngine.firstObject;
+      this.selectedEngineName = selectedEngine[1];
+      this.selectedActions = getActionsByEngineType(this.selectedEngine);
+      this.setSelectedAction(this.selectedActions?.firstObject.actionType);
+    } else {
+      this.selectedEngine = selectedSearchEngines;
+      this.setSelectedAction();
+    }
+  }
+
+  @action
+  setSelectedAction(selectedAction) {
+    if (selectedAction) {
+      this.selectedAction = selectedAction;
+      this.actionParamField = getActionParamByAction(selectedAction);
+    } else {
+      this.selectedActions = getActionsByEngineType('kv');
+      this.actionParamField = getActionParamByAction('find-kv');
+    }
+  }
+
+  @action
+  handleSearch(val) {
+    if (Array.isArray(val)) {
+      this.value = val[0];
+    } else {
+      this.value = val;
+    }
+  }
+
+  @action
+  navigateToPage() {
+    this.router.transitionTo(this.actionParamField.path, this.selectedEngineName, this.value);
```

Comment:
is there a use case where this transition would fail? Thinking about a situation where maybe the value or path are not valid.

Reply:
hmm good question I can't think of a use case where it would fail (may be if the engine page / id) doesn't exist?